commit c5aae75b4588fc0290191ac4135e6838b926119a (refs/changes/19/1619/1)
Author: YangYong3 <yy1111.yang@samsung.com>
Date:   2017-03-07T12:56:55+09:00 (2 years, 7 months ago)
    
    joyent/node-triton#116 `triton package list name=~foo` doesn't seem to filter package ~

diff --git a/lib/papi.js b/lib/papi.js
index 54aac21..75460a6 100644
--- a/lib/papi.js
+++ b/lib/papi.js
@@ -815,7 +815,12 @@ function searchFilter(params, schema) {
             }
         } else {
             if (name !== 'owner_uuids') {
-                constraint = '(' + name + '=' + esc(value) + ')';
+                if (value.toString().charAt(0) === '~') {
+                    value = value.toString().substr(1, value.toString().length);
+                    constraint = '(' + name + '=' + esc(value) + '*)';
+                } else {
+                    constraint = '(' + name + '=' + esc(value) + ')';
+                }
             } else {
                 constraint = '(|(' + name + '=' + esc(value) +
                              ')(!(owner_uuids=*)))';
diff --git a/tools/rsync-to b/tools/rsync-to
new file mode 100755
index 0000000..d2bdcf9
--- /dev/null
+++ b/tools/rsync-to
@@ -0,0 +1,51 @@
+#!/bin/bash
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+#
+# Copyright (c) 2015, Joyent, Inc.
+#
+
+#
+# Rsync the master in this working copy to the install on the given HN.
+#
+
+#set -o xtrace
+set -o errexit
+
+TOP=$(cd $(dirname $0)/../; pwd)
+NODE="root@$1"
+
+if [[ -z "$PAPI_ZONE" ]]; then
+    PAPI_ZONE=$(ssh $NODE "vmadm lookup -1 alias=papi0" 2>/dev/null)
+fi
+echo "PAPI_ZONE: $PAPI_ZONE"
+
+extraOpts=
+if [[ $(uname -s) != "SunOS" ]]; then
+    extraOpts="--exclude *.node --exclude build"
+else
+    # Clean node_modules everytime.
+    ssh $NODE rm -rf /zones/$PAPI_ZONE/root/opt/smartdc/papi/node_modules
+fi
+
+rsync -av ${TOP}/ \
+    $NODE:/zones/$PAPI_ZONE/root/opt/smartdc/papi/ \
+    $extraOpts \
+    --exclude .git/ \
+    --exclude /config.json \
+    --exclude /deps/ \
+    --exclude /doc/ \
+    --exclude /tools/ \
+    --exclude /tmp/ \
+
+state=$(ssh ${NODE} svcs -z ${PAPI_ZONE} -H -o state papi)
+if [[ "$state" == "maintenance" ]]; then
+    ssh ${NODE} svcadm -z ${PAPI_ZONE} clear papi
+else
+    ssh ${NODE} svcadm -z ${PAPI_ZONE} restart papi
+fi
+
